system-addon/test/functional/mochitest/browser_highlights_section.js

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+registerCleanupFunction(async () => {
+  await clearHistoryAndBookmarks();
+});
+
+function refreshHighlightsFeed() {
+  // Toggling the pref will clear the feed cache and force a places query.
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", true);
+}
+
+async function add_highlights_bookmark(count) {
+  const bookmarks = new Array(count).fill(null).map((entry, i) => ({
+    parentGuid: PlacesUtils.bookmarks.unfiledGuid,
+    title: "foo",
+    url: `https://mozilla${i}.com/nowNew`
+  }));
+
+  for (let placeInfo of bookmarks) {
+    await PlacesUtils.bookmarks.insert(placeInfo);
+  }
+
+  // Add bookmark visits
+  for (let placeInfo of bookmarks) {
+    await PlacesTestUtils.addVisits(placeInfo.url);
```

Comment:
Does this second loop need to run strictly after the previous or can the `insert` happen then `addVisits` right away?